internal/cluster/service.go

```diff
@@ -288,6 +289,14 @@ func (s service) DeleteCluster(ctx context.Context, clusterIdentifier Identifier
 		return false, err
 	}
 
+	if !clusterIsReady(c.Status) && !options.Force {
+		return false, NotReadyError{
+			OrganizationID: c.OrganizationID,
+			ID:             c.ID,
+			Name:           c.Name,
+		}
+	}
```

Comment:
I think it would be enough to extend the part above this code like this:

```
	if IsNotFoundError(err) || c.Status == Deleting {
		// Already deleted or being deleted
		return true, nil
	} else if err != nil {
		return false, err
	}
```

What do you think? Unless you want to be able to force delete a cluster that is already in DELETING status.

Reply:
1. We don't want to force delete an already deleting cluster (unless it is stuck, but that is a special case).

2. This could work, I had similar thoughts in mind.

However it is possible this case is already handled elsewhere, because on my tests when I called a delete on my already deleting cluster it ended up returning 202 through the banzai CLI, but it is possible the case is handled either in the CLI or in the API on a higher level. This should be investigated, because it would be better if this was handled in the API than the CLI.

Example:
```
pregnor _ 99895 : PREGNOR-M-82X9 @ 84.3.170.99 . /Users/pregnor [ alpha-cluster / banzaicloud ] 
-> banzai cluster delete --cluster-name pregnor-uv0n4djkvt8eh

# Begin: 2021-11-09T09:39:51.526874000+01:00

Id    Name                   Distribution  Location      Version  CreatorName  CreatedAt             Status    StatusMessage             
6540  pregnor-uv0n4djkvt8eh  eks           eu-central-1  1.18.16  pregnor      2021-11-08T14:32:50Z  DELETING  Termination is in progress
? Do you want to DELETE the cluster? Yes
INFO[0002] Deleting cluster &{202 Accepted 202 HTTP/1.1 1 1 map[Connection:[keep-alive] Content-Length:[0] Date:[Tue, 09 Nov 2021 08:39:53 GMT] Vary:[Accept-Encoding]] 0xc000474d00 0 [] false false map[] 0xc0001b2800 0xc0002cda20} 
Id    Name                   Distribution  Location      Version  CreatorName  CreatedAt             Status    StatusMessage             
6540  pregnor-uv0n4djkvt8eh  eks           eu-central-1  1.18.16  pregnor      2021-11-08T14:32:50Z  DELETING  Termination is in progress

# Executed command: banzai cluster delete --cluster-name pregnor-uv0n4djkvt8eh
# End: 2021-11-09T09:39:53.849148000+01:00
# Duration: 2s322ms274us
# Exit code: 0
```

Reply:
I looked into it and it seems nothing prevents double or any time of deletes, the 202 Accepted gets returned because the cluster delete workflow started again. I just started 3 cluster delete workflows on a cluster, they are listed separately in Cadence.

Reply:
Then IMO we should first guard against multiple deletes probably around here, returning 202 if the cluster is already deleting (and if it fails, we let it fail and the user needs to instruct Pipeline to delete the cluster again after the cluster transitions into an error state).

Reply:
The small modification I proposed in my first response in this thread solves this together with another minor change in internal/cluster/clusterdriver/transport_http.go

Reply:
Let's discuss it under my further comments.